### PR TITLE
feat: datasets list UI

### PR DESCRIPTION
**Pull request by SanketD92**
Issue #, if available:
GALI-1609

Description of changes:
Created a base page for starting DS UI implementation.

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by nguyen102 — COMMENTED** [review]
We probably want to start moving common ui components to `workbench-core`. So I'm thinking the items in `datasets-table-config` and `pages/datasets` should be moved to `datasets-ui`. We then export the page from the project. Then we can import those pages into `swb-ui`. Here's a similar [PR](https://github.com/awslabs/solution-spark-on-aws/pull/460) where `My_Create_React_App` is it's own separate project and it exports a UI component.

**Comment by SanketD92** [comment]
> We probably want to start moving common ui components to `workbench-core`. So I'm thinking the items in `datasets-table-config` and `pages/datasets` should be moved to `datasets-ui`. We then export the page from the project. Then we can import those pages into `swb-ui`. Here's a similar [PR](https://github.com/awslabs/solution-spark-on-aws/pull/460) where `My_Create_React_App` is it's own separate project and it exports a UI component.

There would be a circular dependency for the libraries being used in these files, some of them are placed in `solutions/swb-ui/src/common` and `solutions/swb-ui/src/components` would probably be required to move into a `common-ui` package (scope larger than DS UI implementation).

**Comment by nguyen102** [comment]
> > We probably want to start moving common ui components to `workbench-core`. So I'm thinking the items in `datasets-table-config` and `pages/datasets` should be moved to `datasets-ui`. We then export the page from the project. Then we can import those pages into `swb-ui`. Here's a similar [PR](https://github.com/awslabs/solution-spark-on-aws/pull/460) where `My_Create_React_App` is it's own separate project and it exports a UI component.
> 
> There would be a circular dependency for the libraries being used in these files, some of them are placed in `solutions/swb-ui/src/common` and `solutions/swb-ui/src/components` would probably be required to move into a `common-ui` package (scope larger than DS UI implementation).

Ahh, that's unfortunate. In that case we should create a ticket to move the common components to workbench core. And then we can refactor this PR to include those changes. I'm concerned that not doing that first will just give us more tech debt that we will have to pay back in the future.

**Review by nguyen102 — DISMISSED** [review]

File: workbench-core/datasets-ui/src/datasets-table-config/datasetsColumnDefinitions.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  },
+  {
+    id: 'awsAccountId',
+    header: 'awsAccountId',
+    cell: (e: { awsAccountId: string }) => e.awsAccountId,
+    sortingField: 'awsAccountId'
+  },
+];
+
+export const searchableColumns: string[] = [
```
Comment (bhudnell) [inline workbench-core/datasets-ui/src/datasets-table-config/datasetsColumnDefinitions.tsx:47]: should this be `readonly` as well?

File: workbench-core/datasets-ui/src/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      href: '/datasets'
+    }
+  ];
+
+  // Property and date filter collections
+  const { items, filteredItemsCount, collectionProps, propertyFilterProps } = useCollection(datasets, {
+    filtering: {
+      empty: TableEmptyDisplay(itemType),
+      noMatch: TableNoMatchDisplay(itemType),
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (bhudnell) [inline workbench-core/datasets-ui/src/pages/index.tsx:69]: can these `any`'s be typed?

File: workbench-core/datasets-ui/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  parserOptions: { tsconfigRootDir: __dirname },
+  rules: {
+    // these rules are because the @cloudscape-design/components package doesnt export *Props interfaces in index.js files.
+    // This causes the eslint errors below
+    'import/default': 0,
+    'import/namespace': 0,
+    'import/no-named-as-default': 0,
+    'import/no-named-as-default-member': 0,
+    // disabling this rule because next expects a pages directory which we dont have here
+    '@next/next/no-html-link-for-pages': 0
```
Comment (bhudnell) [inline workbench-core/datasets-ui/.eslintrc.js:15]: are all these necessary in this package?

File: workbench-core/datasets-ui/package.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    "license-checker": "license-checker --onlyAllow 'MIT; Apache-2.0; ISC; BSD'",
+    "lint": "eslint . && rushx pkg-json-lint",
+    "lint:fix": "eslint . --fix",
+    "make-badges": "istanbul-badges-readme --coverageDir=./temp/coverage --exitCode=1",
+    "pkg-json-lint": "npmPkgJsonLint -c ../../.npmpackagejsonlintrc.json .",
+    "sort-package-json": "sort-package-json package.json",
+    "test": "rushx test:only && rushx make-badges",
+    "test:only": "heft test --clean --no-build"
+  },
+  "dependencies": {
```
Comment (bhudnell) [inline workbench-core/datasets-ui/package.json:33]: have you run `rushx depcheck`? It doesnt look like all these dependencies are needed

File: workbench-core/datasets-ui/src/pages/index.tsx (hunk 2 above)
Comment (SanketD92) [reply workbench-core/datasets-ui/src/pages/index.tsx:69]: Yes - for the `filteringFunction` return type (fixed)
No - for the `UseCollectionResult.items`

Comment (bhudnell) [reply workbench-core/datasets-ui/src/pages/index.tsx:69]: It looks like when the `any` is removed from the `items` param it infers `DatasetItem` type. Is that not correct?

File: workbench-core/datasets-ui/src/api/datasets.ts [hunk 5]
```diff
@@ -10,6 +10,7 @@ import { DatasetItem } from '../models/Dataset';
 const useDatasets = (): { datasets: DatasetItem[], areDatasetsLoading: boolean } => {
   const { data, isValidating } = useSWR(() => 'datasets', httpApiGet);
   const datasets: DatasetItem[] = data?.data ?? [];
+  console.log(datasets, isValidating);
```
Comment (bhudnell) [inline workbench-core/datasets-ui/src/api/datasets.ts:13]: should this be removed?

File: workbench-core/datasets-ui/src/pages/index.tsx (hunk 2 above)
Comment (bhudnell) [reply workbench-core/datasets-ui/src/pages/index.tsx:69]: Not sure if you saw my above comment or not

**Review by bhudnell — APPROVED** [review]

**Review by nguyen102 — APPROVED** [review]

**Pull request merged**